### ISSUE TITLE
'failed to unify abilities' when passing through receive

### ISSUE THREAD

**Issue by atacratic** [cmcl/frankjnr#7]
Hi, 

Inspired by logSend in pipes.fk, I tried to write a logReceive, but hit an error.  Here it is with the logging stripped out:

```
receivePassthrough : <Receive String>X -> [Receive String]X
receivePassthrough x              = x
receivePassthrough <receive -> r> = on receive! { s -> receivePassthrough (r s) }
```

The error I hit with that is

`failed to unify abilities Receive ( List Char ) and`

Should it work?

Sorry if this is a silly mistake!

Keep up the good work - frank is very exciting!

Chris

**Comment by slindley** [comment]
Yes, this is a bug.

**Comment by cmcl** [comment]
Hi Chris,

Thanks for reporting this bug! It has now been fixed (3f8c8ee) and a corresponding regression added (tests/should-pass/r7.fk, based on your minimal example -- thanks for that too!)
